Sources/Knit/Module/Container+AbstractRegistration.swift

```diff
@@ -18,6 +18,17 @@ extension Container {
         abstractRegistrations().abstractRegistrations.append(registration)
     }
 
+    /// Register that a service is expected to exist but no implementation is currently available
+    /// If no concrete registration is available then nil will be resolved
```

Comment:
Trying to clarify the behavior a bit here:

```suggestion
    /// Register that a service is expected to exist but no implementation is currently available
    /// The concrete implementation must be registered or the dependency graph is considered invalid
    /// - NOTE: We don't currently support abstract registrations with arguments
    /// As this is an `Optional` Service type this allows special handling of the abstract registration for test environments:
    /// If during testing and no concrete registration is available, then `nil` will be resolved automatically.
```